src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -64,19 +65,38 @@ public ObjectResult(object value)
         {
             var incomingAcceptHeader = HeaderParsingHelpers.GetAcceptHeaders(
                                                 formatterContext.ActionContext.HttpContext.Request.Accept);
-            var sortedAcceptHeaders = SortMediaTypeWithQualityHeaderValues(incomingAcceptHeader)
+            var sortedAcceptHeaderMediaTypes = SortMediaTypeWithQualityHeaderValues(incomingAcceptHeader)
                                         .Where(header => header.Quality != HttpHeaderUtilitites.NoMatch)
                                         .ToArray();
 
+            // By default we want to ignore considering accept headers for content negotiation when
+            // they have a media type like */* in them. Browsers typically have these media types.
+            // In these cases we would want the first formatter in the list of output formatters to
+            // write the response. This default behavior can be changed through options, so checking here.
+            var options = formatterContext.ActionContext.HttpContext
+                                                        .RequestServices
+                                                        .GetRequiredService<IOptions<MvcOptions>>()
+                                                        .Options;
+
             IOutputFormatter selectedFormatter = null;
 
+            bool respectAcceptHeader = true;
+            if (options.RespectBrowserAcceptHeader == false
```

Comment:
This feature mainly started out for browser scenarios where they send */* in requests...but yeah it could be sent from non-browser client too... @yishaigalatzer : should we rename it to RespectAcceptHeader?